src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -1471,22 +1471,30 @@ private static bool AllCharsInVectorAreAscii<T>(Vector128<T> vector)
             else
             {
                 return
-                    Sse41.IsSupported ? Sse41.TestZ(vector.AsInt16(), Vector128.Create((short)-128)) :
+                    Sse41.IsSupported ? Sse41.TestZ(vector.AsUInt16(), Vector128.Create((ushort)0xFF80)) :
                     AdvSimd.Arm64.IsSupported ? AllCharsInUInt64AreAscii(AdvSimd.Arm64.MaxPairwise(vector.AsUInt16(), vector.AsUInt16()).AsUInt64().ToScalar()) :
-                    (vector.AsUInt16() & Vector128.Create((ushort)(ushort.MaxValue - 127))) == Vector128<ushort>.Zero;
+                    (vector.AsUInt16() & Vector128.Create((ushort)0xFF80)) == Vector128<ushort>.Zero;
             }
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool AllCharsInVectorAreAscii<T>(Vector256<T> vector)
             where T : unmanaged
         {
-            Debug.Assert(Avx.IsSupported);
             Debug.Assert(typeof(T) == typeof(byte) || typeof(T) == typeof(ushort));
 
-            return typeof(T) == typeof(byte)
-                ? Avx.TestZ(vector.AsByte(), Vector256.Create((byte)0x80))
-                : Avx.TestZ(vector.AsInt16(), Vector256.Create((short)-128));
+            if (typeof(T) == typeof(byte))
+            {
+                return
+                    Avx.IsSupported ? Avx.TestZ(vector.AsByte(), Vector256.Create((byte)0x80)) :
+                    vector.AsByte().ExtractMostSignificantBits() == 0;
+            }
+            else
+            {
+                return
+                    Avx.IsSupported ? Avx.TestZ(vector.AsUInt16(), Vector256.Create((ushort)0xFF80)) :
+                    (vector.AsUInt16() & Vector256.Create((ushort)0xFF80)) == Vector256<ushort>.Zero;
```

Comment:
subjective: the previous style was easier to read (at least to me)

```suggestion
                return Avx.IsSupported
                    ? Avx.TestZ(vector.AsByte(), Vector256.Create((byte)0x80))
                    : vector.AsByte().ExtractMostSignificantBits() == 0;
            }
            else
            {
                return Avx.IsSupported
                    ? Avx.TestZ(vector.AsUInt16(), Vector256.Create((ushort)0xFF80))
                    : (vector.AsUInt16() & Vector256.Create((ushort)0xFF80)) == Vector256<ushort>.Zero;
```

thank you for adding the support for case when `Avx` is not supported!

Reply:
This method could be annotated as `[BypassReadyToRun]` instead.
The only case when you will hit the non-Avx fallback is when R2R compiled this method without Avx support (no new platforms).

Reply:
> easier to read (at least to me)

Same here. But a few lines above that styles is used to, and for consistency I used that one here too.

> This method could be annotated as `[BypassReadyToRun]` instead.

That attribute is new to me. Thanks for the hint.
I don't know what's better here either, any guidance is welcome.

Reply:
@davidwrighton what would you say the preference is between adding code just to handle the R2R fallback vs. disabling R2R?
In this case at least the fallback is trivial.